sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -237,6 +237,9 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
       OverwritePartitionsDynamicExec(r.table.asWritable, r.options, planLater(query)) :: Nil
 
     case DeleteFromTable(r: DataSourceV2Relation, condition) =>
+      if (SubqueryExpression.hasSubquery(condition)) {
```

Comment:
also see https://github.com/apache/spark/pull/25626/files#diff-fc4a83061d40a242e7f7b729cb8ee870R251

We think that the subquery check is for the current limitation, which should not be treated as a capability check. It's better to put it near the implementation, and should be updated together when we improve the implementation.